### PR TITLE
Support no secret for s3/ceph

### PR DESCRIPTION
**Pull request by tirsen**
This is required if you use EKS ServiceAccount -> IAM role authentication using OIDC.

### Does this PR introduce a user-facing change?:

```release-note
Support no secret for s3/ceph when using OIDC authentication
```

### CODE REVIEW

**Comment by tirsen** [comment]
Using `helm diff install` and removing the `secretName` key:

Before:
```
              - name: AWS_ACCESS_KEY_ID
                valueFrom:
                  secretKeyRef:
-                   name: s3-backup-secret
+                   name:
                    key: access_key
              - name: AWS_SECRET_ACCESS_KEY
                valueFrom:
                  secretKeyRef:
-                   name: s3-backup-secret
+                   name:
                    key: secret_key
```

After this fix:
```
-             - name: AWS_ACCESS_KEY_ID
-               valueFrom:
-                 secretKeyRef:
-                   name: s3-backup-secret
-                   key: access_key
-             - name: AWS_SECRET_ACCESS_KEY
-               valueFrom:
-                 secretKeyRef:
-                   name: s3-backup-secret
-                   key: secret_key
```

File: charts/tidb-backup/templates/scripts/_start_backup.sh.tpl [hunk 1]
```diff
@@ -89,13 +89,24 @@ uploader \
...
+# Once we know there are no more credentials that will be logged we can run with -x
+set -x
+bucket={{ .Values.s3.bucket }}
+
+cat <<EOF > /tmp/rclone.conf
+[aws]
+type = s3
+provider = AWS
+env_auth = true
+region = us-west-2
```
Comment (DanielZhangQD) [inline charts/tidb-backup/templates/scripts/_start_backup.sh.tpl]: This should use the region from values.yaml.

**Comment by tirsen** [comment]
This needs a new build of `tidb-cloud-backup` because it doesn't contain the most recent `rclone`. Web identity provider support was added very recently to `rclone`: https://github.com/rclone/rclone/pull/3837/files

**Comment by DanielZhangQD** [comment]
> https://github.com/rclone/rclone/pull/3837/files

@tennix You have updated the rclone in https://github.com/pingcap/tidb-cloud-backup/pull/7, could you please help update the new image tag of `tidb-cloud-backup` here?

**Comment by tennix** [comment]
> > [#3837 (files)](https://github.com/rclone/rclone/pull/3837/files)
> 
> @tennix You have updated the rclone in [pingcap/tidb-cloud-backup#7](https://github.com/pingcap/tidb-cloud-backup/pull/7), could you please help update the new image tag of `tidb-cloud-backup` here?

Yes, I've already updated to `pingcap/tidb-cloud-backup:20200229`. It should work now.

**Comment by DanielZhangQD** [comment]
@tirsen Could you please help update the image tag of both charts? Thanks!

**Comment by tirsen** [comment]
I tried with the latest image and it still doesn't work. I think we need another fix in rclone: https://github.com/rclone/rclone/pull/4023

**Comment by tirsen** [comment]
Actually I was wrong. I had misconfigured the bucket policy. It works with the latest `tidb-cloud-backup` image!

**Comment by tennix** [comment]
This change might affect the `restore` job, but that could be done in another PR.

**Pull request merged**